tests/src/testCommon.hh

```diff
@@ -201,3 +244,30 @@ template <typename NonLinearOperator, typename FiniteElement>
   }
   return t;
 }
+
+namespace Ikarus {
+  template <typename PowerBasisBasedBasedElement>
+  class AutoDiffBasedFE : public PowerBasisBasedBasedElement {
+  public:
+    using FERequirementType = FErequirements<Eigen::VectorXd>;
+    using LocalView         = typename PowerBasisBasedBasedElement::LocalView;
+    using GridView          = typename PowerBasisBasedBasedElement::GridView;
+    using PowerBasisBasedBasedElement::localView;
+
+    template <typename Basis, typename Material, typename VolumeLoad = std::nullptr_t,
+              typename NeumannBoundaryLoad = std::nullptr_t>
+    requires(Std::is_pointer<VolumeLoad>and Std::is_pointer<NeumannBoundaryLoad>)
+        AutoDiffBasedFE(Basis& globalBasis, const typename LocalView::Element& element, const Material& p_mat,
+                        const BoundaryPatch<GridView>* p_neumannBoundary = nullptr,
+                        const NeumannBoundaryLoad& p_neumannBoundaryLoad = nullptr,
+                        const VolumeLoad& p_volumeLoad                   = nullptr)
+        : PowerBasisBasedBasedElement(globalBasis, element, p_mat, p_neumannBoundary, p_neumannBoundaryLoad,
+                                      p_volumeLoad) {}
+
+  public:
+    template <class ScalarType>
+    ScalarType calculateScalarImpl(const FERequirementType& req, Eigen::VectorX<ScalarType>& dx) const {
```

Comment:
If there is no `calculateVector` or `calculateMatrix` then also the one from `PowerBasisBasedBasedElement` are called.
Thus in the test below you only compare the same functions.

Reply:
Furthermore, the implementation of `calculateScalarImpl` is not useful here.